R/check_year_valid.R

```diff
@@ -24,12 +24,17 @@ check_year_valid <- function(year, type = c(
                                "NSU",
                                "Outpatients",
                                "PIS",
-                               "SDS"
+                               "SDS",
+                               "Sparra-HHG"
                              )) {
   if (year >= "2223" & type == "NSU") {
     return(FALSE)
   }
 
+  if (year >= "2223" & type == "Sparra-HHG") {
+    return(FALSE)
+  }
```

Comment:
SPARRA and HHG have different year limits.

SPARRA starts for 1516 (so end of year 1415), and HHG starts in 1819 (so end of year 1718). HHG was discontinued (#576) so 2223 is the last year we'll get (so 2223 will only ever have start of year, and subsequent years will have none).